### PR TITLE
Small Fixes

### PR DESCRIPTION
**Pull request by CarrotManMatt**
* Updates dependencies
* Improves selected linting rules
* Improves class names
* Improves command names & descriptions
* Improves custom exception inheritance/abstract-base-class usage
* Fix references of "the bot" into "TeX-Bot"
* Ensures all loggers are annotated as `Final`
* Improves autocomplete function type annotations
* Improves formatting of channel & role names (within documentation & response messages)
* Improves imports & `__all__` class & function exports
* Fixes spelling & grammar mistakes
* Improves the usage of `Final` constants and their capitalisation
* Improves command response messages
* Improves autocomplete exception handling
* Temporarily improves settings names (this will be overhauled in #221)

### CODE REVIEW

File: cogs/strike.py [hunk 1]
```diff
@@ -473,56 +483,48 @@ async def _confirm_manual_add_strike(self, strike_user: discord.User | discord.M
...
             await DiscordMemberStrikes.objects.aget_or_create(
                 discord_id=strike_user.id,
             )
         )[0]
 
-        strikes_out_of_sync_with_ban: bool = bool(
+        STRIKES_OUT_OF_SYNC_WITH_BAN: Final[bool] = bool(
             (action != discord.AuditLogAction.ban and member_strikes.strikes >= 3)
-            or (action == discord.AuditLogAction.ban and member_strikes.strikes > 3),
+            or (action == discord.AuditLogAction.ban and member_strikes.strikes > 3)  # noqa: COM812
```
Comment (MattyTheHacker) [inline cogs/strike.py:525]: why are we removing the trailing commas here? (and elsewhere)

File: exceptions/does_not_exist.py [hunk 2]
```diff
@@ -207,26 +193,21 @@ class MemberRoleDoesNotExistError(RoleDoesNotExistError):
...
-    def DEPENDENT_COMMANDS(cls) -> frozenset[str]:  # noqa: N802,N805
-        """
-        The set of names of bot commands that require this Discord entity.
-
-        This set being empty could mean that all bot commands require this Discord entity,
-        or no bot commands require this Discord entity.
-        """  # noqa: D401
+    @override
+    def DEPENDENT_COMMANDS(cls) -> frozenset[str]:  # noqa: N805
         # noinspection SpellCheckingInspection
```
Comment (MattyTheHacker) [inline exceptions/does_not_exist.py:204]: what's the logic behind removing all this documentation (not this one specififcally but all of it)

**Review by MattyTheHacker — CHANGES_REQUESTED** [review]

File: exceptions/does_not_exist.py (hunk 2 above)
Comment (CarrotManMatt) [reply exceptions/does_not_exist.py:204]: Due to the use of the `@override` decorator the documentation should get inherited from the parent class. In all these cases the child class docstrings were replicas of the parent class's docstrings so just caused more places for updates to need to be made. I'm not sure about VSCode, but certainly when PyCharm identifies an `@override`n method, hovering over that method will display the docstring of the parent class.

File: cogs/remind_me.py [hunk 3]
```diff
@@ -27,14 +28,14 @@
...
+logger: Final[Logger] = logging.getLogger("TeX-Bot")
 
 
 class RemindMeCommandCog(TeXBotBaseCog):
-    """Cog class that defines the "/remindme" command and its call-back method."""
+    """Cog class that defines the "/remind-me" command and its call-back method."""
 
     @staticmethod
-    async def autocomplete_get_delays(ctx: TeXBotAutocompleteContext) -> set[str]:  # noqa: C901, PLR0912, PLR0915
+    async def autocomplete_get_delays(ctx: TeXBotAutocompleteContext) -> Set[discord.OptionChoice] | Set[str]:  # noqa: C901,PLR0912,PLR0915,E501
```
[the remark this answers is not in the document]
Comment (CarrotManMatt) [reply cogs/remind_me.py:38]: Although there is no formal rule for how to annotate multiple `# noqa` comments in a list, all the examples I have seen from ruff & mypy's documentation suggest to just separate with commas and not spaces to reduce line length. Especially when they're more focused for machine reading rather than developer reading.

File: exceptions/does_not_exist.py (hunk 2 above)
Comment (MattyTheHacker) [reply exceptions/does_not_exist.py:204]: ah yeah vsc does the same, fairs

Comment (CarrotManMatt) [reply exceptions/does_not_exist.py:204]: Phew. That wouldve been more irritating if we had to accommodate a missing feature with worse code.

File: cogs/strike.py (hunk 1 above)
Comment (MattyTheHacker) [reply cogs/strike.py:525]: I think I'd rather not have the comma but then having to have the ignore comment looks just as stupid so tbh I don't know which is worse

File: cogs/remind_me.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MattyTheHacker) [reply cogs/remind_me.py:38]: hmmm, I do think in general it's easier to read if they have the space so I think i'd rather that, but in any case it definitely needs to be consistent so if you wanted to do it without the spaces you'd need to go round remove them all

[the remark this answers is not in the document]
Comment (CarrotManMatt) [reply cogs/remind_me.py:38]: Cool, I'll remove those changes from this PR and we can formalise on having spaces.

I'm planning on adding rules like this to a custom linter based on a flake8 plugin. We'll use ruff for the common linting rules because it's written in rust so is much faster and the ones that aren't handled by that (like command names being hyphenated etc) can be implemented in my custom linter. (This will be a separate repo project)

File: cogs/strike.py (hunk 1 above)
Comment (CarrotManMatt) [reply cogs/strike.py:525]: I prefer the ignore, purely because it gets coloured as a darker gray in my editor so hides more into the background. It also makes it clear that the thing on the end there is purely about formatting because it is a comment & not actual code.

**Comment by MattyTheHacker** [comment]
Also why are we changing settings names just to change it again in the config commit? surely it's easier to leave it as is for now then make it better with the config pr

**Comment by CarrotManMatt** [comment]
The changed settings are using the names that will be used after the config update. Just thought that some of the changes matched with some of the fixes here. I suppose there's no difference as to whether they get changed here or in the config.

So the formal question: is it an issue enough that you would like me to remove those changes from this PR?

**Comment by MattyTheHacker** [comment]
If you can gimme a list of the ones that need changing in the prod config and what they need changing to then it can stay in, not that bothered

**Comment by CarrotManMatt** [comment]
Oh sorry. I didn't make that clear. The external side names I left the same for this exact reason. So the deployment config will only need changing once, as soon as #221 is merged. The changes here are just the internal names to adhere to the format rules specified in the comment at the top of this PR.

TL;DR: The deployment environment variables should not need to be changed for this PR.

**Comment by MattyTheHacker** [comment]
oh sick no problem at all then

**Review by MattyTheHacker — APPROVED** [review]
